### PR TITLE
Tag containers with WeaveDNS hostnames.

### PR DESCRIPTION
**Pull request by tomwilkie**

### CODE REVIEW

**Comment by tomwilkie** [comment]
I think the test failures are related to the github outage.  Will rerun later.

File: probe/overlay/weave.go [hunk 1]
```diff
@@ -26,27 +29,37 @@ const (
 // overlay -- though I'm not sure what that would look like in practice right
 // now.
 type Weave struct {
-	url string
+	url   string
+	scope string
```
Comment (peterbourgon) [inline probe/overlay/weave.go]: s/scope/hostID/g

File: render/detailed_node.go [hunk 2]
```diff
@@ -296,6 +297,7 @@ func containerOriginTable(nmd report.NodeMetadata, addHostTag bool) (Table, bool
 	rows := []Row{}
 	for _, tuple := range []struct{ key, human string }{
 		{docker.ContainerID, "ID"},
+		{overlay.WeaveDNSHostname, "Weave DNS Hostname:"},
```
Comment (peterbourgon) [inline render/detailed_node.go]: Subtract the colon

File: probe/overlay/weave.go [hunk 3]
```diff
@@ -72,8 +85,26 @@ func (w Weave) update() (weaveStatus, error) {
...
+	status, err := w.update()
+	if err != nil {
+		return r, nil
+	}
+
+	for _, entry := range status.DNS.Entries {
+		if entry.Tombstone > 0 {
+			continue
+		}
+		nodeid := report.MakeContainerNodeID(w.scope, entry.ContainerID)
```
Comment (peterbourgon) [inline probe/overlay/weave.go]: nodeID

File: probe/overlay/weave_test.go [hunk 4]
```diff
@@ -16,43 +16,81 @@ func TestWeaveTaggerOverlayTopology(t *testing.T) {
...
-			report.MakeOverlayNodeID(mockWeavePeerName): report.MakeNodeMetadataWith(map[string]string{
-				overlay.WeavePeerName:     mockWeavePeerName,
-				overlay.WeavePeerNickName: mockWeavePeerNickName,
-			}),
-		},
-	}), have.Overlay; !reflect.DeepEqual(want, have) {
-		t.Error(test.Diff(want, have))
+
+	{
+		nodeid := report.MakeContainerNodeID(mockHostID, mockContainerID)
```
Comment (peterbourgon) [inline probe/overlay/weave_test.go]: nodeID

**Comment by tomwilkie** [comment]
@peterbourgon PTAL

File: probe/overlay/weave.go [hunk 5]
```diff
@@ -72,8 +85,26 @@ func (w Weave) update() (weaveStatus, error) {
...
+	status, err := w.update()
+	if err != nil {
+		return r, nil
+	}
+
+	for _, entry := range status.DNS.Entries {
+		if entry.Tombstone > 0 {
+			continue
+		}
+		nodeid := report.MakeContainerNodeID(w.hostID, entry.ContainerID)
```
Comment (peterbourgon) [inline probe/overlay/weave.go]: nodeID

**Comment by peterbourgon** [comment]
Are we sure it's OK to make an HTTP call from the probe to the Weave router with every Tag and Report, i.e. once per publish interval?

**Comment by tomwilkie** [comment]
Yes I'm pretty confident this is safe; everything we're querying is in memory, and the weave routers deals with many more QPS on other interfaces (ie DNS).  Lets put it another way: I'm more confident this is safe than the amount of queries we do on the docker daemon!

**Comment by peterbourgon** [comment]
:+1:

**Pull request merged**